stages/org.osbuild.vagrant

```diff
@@ -23,20 +151,48 @@ def main(tree, options, inputs):
     # https://github.com/vagrant-libvirt/vagrant-libvirt/tree/master/example_box
     target = os.path.join(tree, "box.img")
     provider = options["provider"]
+    if provider == "virtualbox":
+        # the box has to be named .vmdk because Virtualbox does not detet the mimetype..
+        target = os.path.join(tree, "box.vmdk")
     subprocess.run(["cp", "-a", "--reflink=auto", source, target], check=True)
+    vagrant_content = ""
 
     metadata = {"provider": options["provider"]}
     if provider == "libvirt":
         metadata["format"] = "qcow2"
         # virtual image size as rounded numbwr
         data = json.loads(subprocess.check_output(["qemu-img", "info", "--output", "json", target]))
         metadata["virtual_size"] = data['virtual-size'] // 1000 ** 3
+        vagrant_content = """
+config.vm.provider :libvirt do |libvirt|
+  libvirt.driver = "kvm"
+end
+        """
+
+    if provider == "virtualbox":
+        # https://github.com/mirror/vbox/blob/b9657cd5351cf17432b664009cc25bb480dc64c1/src/VBox/Main/src-server/HostImpl.cpp#L3267
+        # VirtualBox-6.1.12 src/VBox/NetworkServices/Dhcpd/Config.cpp line 276
+        mac_address = "080027"
+        for _ in range(0, 3):
+            mac_address += "".join(random.sample("0123456789abcdef", 2))
+        vagrant_content = f"config.vm.base_mac = \"{mac_address}\""
+
+        data = json.loads(subprocess.check_output(["qemu-img", "info", "--output", "json", target]))
+        xml = BOX_XML.format(
+            MAC_ADDRESS=mac_address,
+            MACHINE_UUID=str(
+                uuid.uuid4()),
+            DISK_UUID=str(
+                uuid.uuid4()),
+            DISK_CAPACITY=data['virtual-size'] //
+            1000 ** 3,
+            DISK_NAME=os.path.basename(target),
+            OS_TYPE="ArchLinux_64")
```

Comment:
:thinking: should probably be configurable or automatically set to something sane.